src/plugins/controls/server/control_types/options_list/options_list_queries.ts

```diff
@@ -0,0 +1,174 @@
+import { getFieldSubtypeNested } from '@kbn/data-views-plugin/common';
+
+import { OptionsListRequestBody } from '../../../common/control_types/options_list/types';
+
+export interface OptionsListAggregationBuilder {
+  buildAggregation: (req: OptionsListRequestBody) => unknown;
+  parse: (response: SearchResponse) => string[];
+}
+
+/**
+ * Validation aggregations
+ */
+export const getValidationAggregationBuilder: () => OptionsListAggregationBuilder = () => ({
+  buildAggregation: ({ selectedOptions, fieldName }: OptionsListRequestBody) => {
+    const selectedOptionsFilters = selectedOptions?.reduce((acc, currentOption) => {
+      acc[currentOption] = { match: { [fieldName]: currentOption } };
+      return acc;
+    }, {} as { [key: string]: { match: { [key: string]: string } } });
+
+    return selectedOptionsFilters && !isEmpty(selectedOptionsFilters)
+      ? {
+          filters: {
+            filters: selectedOptionsFilters,
+          },
+        }
+      : undefined;
+  },
+  parse: (rawEsResult) => {
+    const rawInvalidSuggestions = get(rawEsResult, 'aggregations.validation.buckets') as {
+      [key: string]: { doc_count: number };
+    };
+    return rawInvalidSuggestions && !isEmpty(rawInvalidSuggestions)
+      ? Object.entries(rawInvalidSuggestions)
+          ?.filter(([, value]) => value?.doc_count === 0)
+          ?.map(([key]) => key)
+      : [];
+  },
+});
+
+/**
+ * Suggestion aggregations
+ */
+export const getSuggestionAggregationBuilder = ({
+  fieldSpec,
+  textFieldName,
+  searchString,
+}: OptionsListRequestBody) => {
+  if (textFieldName && fieldSpec?.aggregatable && searchString) {
+    return suggestionAggSubtypes.keywordAndText;
+  }
+  if (fieldSpec?.type === 'boolean') {
+    return suggestionAggSubtypes.boolean;
+  }
+  if (fieldSpec && getFieldSubtypeNested(fieldSpec)) {
+    return suggestionAggSubtypes.subtypeNested;
+  }
+  return suggestionAggSubtypes.keywordOnly;
+};
+
+const getEscapedQuery = (q: string = '') =>
+  q.replace(/[.?+*|{}[\]()"\\#@&<>~]/g, (match) => `\\${match}`);
+
+const suggestionAggSubtypes: { [key: string]: OptionsListAggregationBuilder } = {
+  /**
+   * the "Keyword only" query / parser should be used when the options list is built on a field which has only keyword mappings.
+   */
+  keywordOnly: {
+    buildAggregation: ({ fieldName, searchString }: OptionsListRequestBody) => ({
+      terms: {
+        field: fieldName,
+        include: `${getEscapedQuery(searchString)}.*`,
+        execution_hint: 'map',
+        shard_size: 10,
+      },
+    }),
+    parse: (rawEsResult) =>
+      get(rawEsResult, 'aggregations.suggestions.buckets')?.map(
+        (suggestion: { key: string }) => suggestion.key
+      ),
+  },
+
+  /**
+   * the "Keyword and text" query / parser should be used when the options list is built on a multi-field which has both keyword and text mappings. It supports case-insensitive searching
+   */
+  keywordAndText: {
```

Comment:
I tested this in two ways. The first was with the typical `text` parent and `keyword` child - for this, I simply used the `customer_first_name.keyword` field from the ecommerce sample data and, as expected, search **was** case sensitive - if I searched for `a`, I got the exact same results as when I searched for `A`.

The second test was for a `keyword` parent and a `text` child - this is definitely less common since most documentation uses the previous mapping (and the previous mapping is also the default for `string` fields). However, since it's technically **possible** to have this mapping, I wanted to make sure it was still supported. To do this, I reindexed the sample ecommerce data with new mapping where `customer_last_name` looked like this:
```
... "customer_last_name": {
  "type": "keyword",
  "fields": {
    "text": {
      "type": "text"
    }
  }
}, ...
```
Then, I made a control using this new data view + field combo and, sure enough, search was still case sensitive - woohoo! Searching for both `per` and `Per` gave me `["Perkins", "Perry"]` as suggestions :)